src/App.tsx

```diff
@@ -17,18 +71,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedOption={selectedOption}
+                handleSelectChange={setSelectedOption}
+                query={query}
+                handleQueryChange={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodoId={selectedTodo?.id}
+                  onSelect={setSelectedTodo}
+                  key={selectedTodo?.id}
```

Comment:
Generating key on render is not recommended as it can lead to unexpected behavior. In this case, you're using `selectedTodo?.id` as a key, which will be regenerated every time the `selectedTodo` changes. This can lead to performance issues and unexpected behavior in your React components. Here is a resource that explains this issue in more detail: [link](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052)

Reply:
is'nt it what you told me to do earlier???